libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -170,7 +170,16 @@ void ExpressionCompiler::appendStateVariableAccessor(VariableDeclaration const&
 			// pop offset
 			m_context << Instruction::POP;
 			utils().copyToStackTop(paramTypes.size() - i + 1, 1);
-			ArrayUtils(m_context).accessIndex(*arrayType);
+
+			ArrayUtils(m_context).retrieveLength(*arrayType, 1);
```

Comment:
So for a dynamically sized calldata type...
Stack pre is: ``ref length index``
Then ``retrieveLength`` will put in a ``dupInstruction(1+ _stackDepth)`` so ``dupInstruction(2)``, so ``DUP2``....
Which is ``length``...
So yeah, for the record: I checked, this seems fine.